### PR TITLE
refactor: metadata and metablock

### PR DESCRIPTION
**Pull request by Xynnn007**
Divide metadata into object-safe part and object-unsafe part,
delete generics in Metadata

refer to https://github.com/in-toto/in-toto-rs/issues/33

### CODE REVIEW

File: src/models/metadata.rs [hunk 1]
```diff
@@ -1,211 +1,150 @@
...
-            .into_iter()
-            .map(|(_k, v)| v)
-            .collect::<Vec<_>>();
-        signatures.sort_unstable_by(|a, b| a.key_id().cmp(b.key_id()));
-
-        Metablock {
-            signatures,
-            metadata: self.metadata,
-            _marker: PhantomData,
+    /// Auto deserialize for MetadataWrapper by any possiable metadata.
```
[the remark this answers is not in the document]
Comment (Xynnn007) [reply src/models/metadata.rs:78]: XD

**Review by SantiagoTorres — COMMENTED** [review]
Quick question. Since now to_bytes performs canonicalization. This means all serialized metadata is *also* canonicalized right?

**Comment by Xynnn007** [comment]
> Quick question. Since now to_bytes performs canonicalization. This means all serialized metadata is _also_ canonicalized right?

Here are two different interfaces to serialize metadata
* `serde_json::to_string(&metadata)` and all the `json_serde` operations: This will not involve any canonicalization
* `metadata.to_bytes()`: This will call `canonicalize()` upon the result of serialized metadata, as _you've mentioned_

**Comment by SantiagoTorres** [comment]
LGTM! thank you @Xynnn007 !

**Pull request merged**